app/src/main/java/ar/com/wolox/android/cookbook/twitterlogin/TwitterLoginRecipeFragment.kt

```diff
@@ -0,0 +1,121 @@
+package ar.com.wolox.android.cookbook.twitterlogin
+
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import android.net.Uri
+import android.view.View
+import android.widget.Toast
+import androidx.fragment.app.FragmentActivity
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.twitterlogin.model.YoutubeEmailResponse
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import com.twitter.sdk.android.core.Callback
+import com.twitter.sdk.android.core.TwitterSession
+import com.twitter.sdk.android.core.models.User
+import kotlinx.android.synthetic.main.fragment_twitter_login.*
+
+class TwitterLoginRecipeFragment : WolmoFragment<TwitterLoginRecipePresenter>(), TwitterLoginRecipeView {
+
+    override fun layout(): Int = R.layout.fragment_twitter_login
+
+    override fun init() {
+    }
+
+    override fun setListeners() {
+        vDefaultLoginBtn.setOnClickListener {
+            presenter.onDefaultBtnRequest()
+        }
+
+        vCustomLoginBtn.setOnClickListener {
+            presenter.onCustomBtnRequest()
+        }
+
+        vGetProfileBtn.setOnClickListener {
+            presenter.onImageRequest()
+        }
+
+        vLogoutBtn.setOnClickListener {
+            presenter.onLogoutRequest()
+        }
+    }
+
+    override fun getActivityContext(): FragmentActivity? {
+        return activity
+    }
+
+    override fun isNetworkAvailable(): Boolean {
+        val connectivityManager = context?.getSystemService(Context.CONNECTIVITY_SERVICE)
+        return if (connectivityManager is ConnectivityManager) {
+            val networkInfo: NetworkInfo? = connectivityManager.activeNetworkInfo
+            networkInfo?.isConnected ?: false
+        } else false
+    }
+
+    override fun showNetworkUnavailableError() {
+        Toast.makeText(context, getString(R.string.twitter_error_network_unavailable), Toast.LENGTH_LONG).show()
+    }
+
+    override fun setLoginCallback(callback: Callback<TwitterSession>) {
+        vDefaultLoginBtn.callback = callback
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        vDefaultLoginBtn.onActivityResult(requestCode, resultCode, data)
+        presenter.onActivityResult(requestCode, requestCode, data)
```

Comment:
I believe that here onActivityResult has finished, so we can change method to `onActivityResultFinished()`. I'm wrong with that?